src/MergeContext.js

```diff
@@ -997,16 +1002,13 @@ class PullRequest {
     async _getStagingStatuses() {
         const combinedStagingStatuses = await GH.getStatuses(this._stagedSha());
         const genuineStatuses = combinedStagingStatuses.statuses.filter(st => !st.description.endsWith(Config.copiedDescriptionSuffix()));
-        assert(genuineStatuses.length <= Config.stagingChecks());
-        let statusChecks = new StatusChecks(Config.stagingChecks(), "Staging", this._contextsRequiredByGitHubConfig);
-        // all genuine checks, except for PR approval, are 'required'
-        // PR approval may be either 'required' or 'optional' (depending on the GitHub settings)
+        let statusChecks = new StatusChecks("Staging", this._contextsRequiredByGitHubConfigStaging);
         for (let st of genuineStatuses) {
```

Comment:
If you agree with it, please leave this code polishing comment for a followup PR to address:

```suggestion
        // TODO: Move this loop inside StatusChecks constructor by adding
        // a third constructor parameter to pass genuineStatuses (here) and
        // combinedPrStatus.statuses (in similar _getPrStatuses() code).
        for (let st of genuineStatuses) {
```

If you agree with the "move this loop inside StatusChecks constructor" part but see problems with "by ..." part than just add the first part:

```suggestion
        // TODO: Move this loop inside StatusChecks constructor to reuse
        // it here and in similar _getPrStatuses() code.
        for (let st of genuineStatuses) {
```